src/loader/bgw_message_queue.h

```diff
@@ -0,0 +1,41 @@
+#ifndef BGW_MESSAGE_QUEUE_H
+#define BGW_MESSAGE_QUEUE_H
+
+#include <postgres.h>
+#include <storage/dsm.h>
+
+
+typedef enum tsbgwMessageType
+{
+	STOP = 0,
+	START,
+	RESTART
+}			tsbgwMessageType;
+
+typedef struct tsbgwMessage
+{
+	tsbgwMessageType message_type;
```

Comment:
Optional: Given the ack_dsm_handle and offset stuff I feel increasingly uncomfortable with this structure being public. Would be better to have a public tsbgMessage type that only has tsbgwMessageType message_type; and a private tsbgMessageEnvelope type that contains a message and the rest of these fields that's only in the .c file.

Reply:
Those fields need to be used by the launcher, so I'm not quite sure how that would work. bgw_message_queue.h is public only inside the loader, the entire functionality to the outside world is in timescale_bgw.h. My impression is we have the choice that either we have the queue do all the work of interpreting and acting on messages, or these fields go in the .h . I suppose the other way of thinking about this would be to